avocado/core/nrunner.py

```diff
@@ -557,29 +557,32 @@ class PythonUnittestRunner(BaseRunner):
 
      * kwargs: not used
     """
-    @staticmethod
-    def _uri_to_unittest_name(uri):
+
+    @property
+    def unittest_name(self):
+        """Convert a test reference (uri) to an unittest name reference."""
+        uri = self.runnable.uri
+        if not uri:
```

Comment:
`kind` and `uri` are required arguments of a `Runnable`.  Granted, you can give it a wrong type like `None` or a bool, but I don't think we should be doing type checking, and the following line `if ':' in uri` would already cover empty strings.

Reply:
I can't reproduce now, but I went to some cases were self.runnable.uri was None. And `if ':' in None` would raise a `TypeError` exception. But yes, makes sense.